recipe/thirdparty/providers/apple.go

```diff
@@ -20,123 +20,93 @@ import (
+				return tpmodels.ProviderConfigForClient{}, err
 			}
 
-			authorisationRedirectURL := "https://appleid.apple.com/auth/authorize"
-			scopes := []string{"email"}
-			if config.Scope != nil {
-				scopes = config.Scope
+			if len(config.Scope) == 0 {
+				config.Scope = []string{"openid", "email"}
 			}
 
-			var additionalParams map[string]interface{} = nil
-			if config.AuthorisationRedirect != nil && config.AuthorisationRedirect.Params != nil {
-				additionalParams = config.AuthorisationRedirect.Params
+			if config.ClientSecret == "" {
+				clientSecret, err := getClientSecret(config.ClientID, config.AdditionalConfig["clientSecret"].(map[string]interface{}))
+				if err != nil {
+					return tpmodels.ProviderConfigForClient{}, err
+				}
+				config.ClientSecret = clientSecret
 			}
 
-			authorizationRedirectParams := map[string]interface{}{
-				"scope":         strings.Join(scopes, " "),
-				"response_mode": "form_post",
-				"response_type": "code",
-				"client_id":     config.ClientID,
-			}
-			for key, value := range additionalParams {
-				authorizationRedirectParams[key] = value
-			}
+			return config, err
+		}
 
-			return tpmodels.TypeProviderGetResponse{
-				AccessTokenAPI: tpmodels.AccessTokenAPI{
-					URL:    accessTokenAPIURL,
-					Params: accessTokenAPIParams,
-				},
-				AuthorisationRedirect: tpmodels.AuthorisationRedirect{
-					URL:    authorisationRedirectURL,
-					Params: authorizationRedirectParams,
-				},
-				GetProfileInfo: func(authCodeResponse interface{}, userContext supertokens.UserContext) (tpmodels.UserInfo, error) {
-					claims, err := verifyAndGetClaimsAppleIdToken(authCodeResponse.(map[string]interface{})["id_token"].(string), api.GetActualClientIdFromDevelopmentClientId(config.ClientID))
-					if err != nil {
-						return tpmodels.UserInfo{}, err
-					}
-
-					var email string
-					var isVerified bool
-					var id string
-					for key, val := range claims {
-						if key == "sub" {
-							id = val.(string)
-						} else if key == "email" {
-							email = val.(string)
-						} else if key == "email_verified" {
-							isVerified = val.(string) == "true"
-						}
-					}
-					return tpmodels.UserInfo{
-						ID: id,
-						Email: &tpmodels.EmailStruct{
-							ID:         email,
-							IsVerified: isVerified,
-						},
-					}, nil
-				},
-				GetClientId: func(userContext supertokens.UserContext) string {
-					return config.ClientID
-				},
-				GetRedirectURI: func(userContext supertokens.UserContext) (string, error) {
-					supertokens, err := supertokens.GetInstanceOrThrowError()
-					if err != nil {
-						return "", err
-					}
-					return supertokens.AppInfo.APIDomain.GetAsStringDangerous() + supertokens.AppInfo.APIBasePath.GetAsStringDangerous() + "/callback/apple", nil
-				},
-			}
-		},
-		IsDefault: config.IsDefault,
+		if oOverride != nil {
+			provider = oOverride(provider)
+		}
+		return provider
 	}
+
+	return NewProvider(input)
 }
```

Comment:
Copy profile info from redirect uri info to raw user info => from id token

Reply:
Done